src/cargo/core/shell.rs

```diff
@@ -119,199 +91,82 @@ impl MultiShell {
         where F: FnMut(&mut MultiShell) -> CargoResult<()>
     {
         match self.verbosity {
-            Verbose => Ok(()),
+            Verbosity::Verbose => Ok(()),
             _ => callback(self)
         }
     }
 
     pub fn error<T: fmt::Display>(&mut self, message: T) -> CargoResult<()> {
-        self.err().say_status("error:", message, RED, false)
+        self.print(&"error:", &message, Red, false)
     }
 
     pub fn warn<T: fmt::Display>(&mut self, message: T) -> CargoResult<()> {
         match self.verbosity {
-            Quiet => Ok(()),
-            _ => self.err().say_status("warning:", message, YELLOW, false),
+            Verbosity::Quiet => Ok(()),
+            _ => self.print(&"warning:", &message, Yellow, false),
         }
     }
 
     pub fn set_verbosity(&mut self, verbosity: Verbosity) {
         self.verbosity = verbosity;
     }
 
-    pub fn set_color_config(&mut self, color: Option<&str>) -> CargoResult<()> {
+    pub fn verbosity(&self) -> Verbosity {
+        self.verbosity
+    }
+
+    pub fn set_color_choice(&mut self, color: Option<&str>) -> CargoResult<()> {
         let cfg = match color {
-            Some("auto") => Auto,
-            Some("always") => Always,
-            Some("never") => Never,
+            Some("always") => Choice::Always,
+            Some("never") => Choice::Never,
 
-            None => Auto,
+            Some("auto") |
+            None => Choice::CargoAuto,
 
             Some(arg) => bail!("argument for --color must be auto, always, or \
                                 never, but found `{}`", arg),
-        };
-        self.out.set_color_config(cfg);
-        self.err.set_color_config(cfg);
-        Ok(())
-    }
-
-    pub fn get_verbose(&self) -> Verbosity {
-        self.verbosity
+       };
```

Comment:
Dammit, I've pressed "Add single comment" instead of "Start a review" again :( 

What is more convenient for you as an author of pull request: a ton of single comments, or a single review?